syntax/types.rs

```diff
@@ -116,7 +116,7 @@ impl<'a> Types<'a> {
                     rust.insert(ident);
                 }
                 Api::CxxFunction(efn) | Api::RustFunction(efn) => {
-                    let ident = &efn.ident;
+                    let ident = if let Some(alias) = &efn.alias { alias } else { &efn.ident };
                     if !function_names.insert((&efn.receiver, ident)) {
                         duplicate_name(cx, efn, ident);
                     }
```

Comment:
As implemented in this PR, the `alias` of an extern Rust function is its C++ name and the `alias` of an extern C++ function is its Rust name. So this check isn't correct:

- It is going to have false positives on extern Rust functions having the same C++ name, which we should allow because C++ has function overloading.
- It is going to have false negatives on an extern C++ function with alias *f* colliding with an extern Rust function named *f* if the Rust one also has an alias.
- It is going to have false positives on the Rust name of a C++ function being the same as the C++ name of a Rust function, which should in general have no bearing on each other.